test/mechanics/cell/Jacobian.cpp

```diff
@@ -59,6 +61,42 @@ BOOST_AUTO_TEST_CASE(Jacobian1Din2D)
     }
 }
 
+//! @brief calculate the area of a triangle (points a,b,c) element based on ||J|| and check vs. `correctArea`
+void CheckJacobian2Din3D(Eigen::Vector3d a, Eigen::Vector3d b, Eigen::Vector3d c, double correctArea)
+{
+    Eigen::MatrixXd B = NuTo::ShapeFunctions2D::DerivativeShapeFunctionsTriangleOrder1(Eigen::Vector2d::Zero());
+    Eigen::VectorXd coordinates = Eigen::VectorXd(9);
+    coordinates << a.x(), a.y(), a.z(), b.x(), b.y(), b.z(), c.x(), c.y(), c.z();
+
+    NuTo::Jacobian jacobian(coordinates, B, 3);
+
+    double ratioAreaRealToAreaNatural = jacobian.Det();
+    double areaNatural = 0.5; // triangle (0,0) ; (1,0) ; (0,1)
+    double calculatedArea = ratioAreaRealToAreaNatural * areaNatural;
+
+    BOOST_CHECK_CLOSE(correctArea, calculatedArea, 1.e-10);
+}
+
+BOOST_AUTO_TEST_CASE(Jacobian2Din3D)
+{
+    // some manual test cases
+    CheckJacobian2Din3D(Eigen::Vector3d(0, 0, 0), Eigen::Vector3d(1, 0, 0), Eigen::Vector3d(0, 1, 0), 0.5);
+    CheckJacobian2Din3D(Eigen::Vector3d(0, 0, 0), Eigen::Vector3d(5, 0, 0), Eigen::Vector3d(0, 0, 7), 5. * 7 / 2);
+    CheckJacobian2Din3D(Eigen::Vector3d(0, 4, 0), Eigen::Vector3d(0, 0, 3), Eigen::Vector3d(0, 4, 3), 3. * 4 / 2);
+
+    // some random test cases
+    for (int i = 0; i < 100; ++i)
+    {
```

Comment:
Randomized tests might be nice, but I don't think it is going to catch any problems that the above lines don't catch. Maybe we could have a couple of tests with problematic input and how it is being handled: "Triangle" with two points that coincide, a quad with negative jacobian, this sort of thing.

Reply:
You are right, these random tests provide no benefit. But I think the jacobian itself should not check invalid inputs.

Reply:
Fair enough.